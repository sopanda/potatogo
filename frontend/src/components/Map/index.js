export { Map as default } from './Map';
