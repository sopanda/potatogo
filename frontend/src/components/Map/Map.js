import React, { forwardRef, memo } from 'react';
import { Map as LeafletMap, TileLayer } from 'react-leaflet';
import { Markers } from 'components/Markers';
import styled from 'styled-components';
import { MAPS_URL } from 'constants/maps';

const MapWrapper = styled(LeafletMap)`
  position: relative;
  width: 100%;
  min-height: 500px;

  & .leaflet-control-container,
  & .leaflet-routing-container-hide {
    display: none !important;
  }
`;

export const Map = memo(
  forwardRef((props, ref) => {
    const { latitude, longitude, children, type, addMarker, markers } = props;
    return (
      <MapWrapper
        center={[latitude, longitude]}
        zoom={15}
        attributionControl
        zoomControl
        doubleClickZoom
        scrollWheelZoom
        dragging
        animate
        easeLinearity={0.35}
        onClick={e => type === 'create' && addMarker(e)}
        ref={ref}
      >
        <TileLayer url={MAPS_URL.WIKIMEDIA} />
        {markers && type === 'home' && <Markers items={markers} />}
        {type === 'create' && children}
      </MapWrapper>
    );
  })
);
