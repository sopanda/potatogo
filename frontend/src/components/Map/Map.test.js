import React from 'react';
import { shallow } from 'enzyme';
import Map from './';

describe('Map component tests', () => {
  let component;
  let coords = {
    latitude: 50.041187,
    longitude: 21.999121
  };
  beforeEach(() => {
    component = shallow(<Map coords={coords} />);
  });

  it('should render correctly with props', () => {
    expect(component).toMatchSnapshot();
  });

  it('should have props with location', () => {
    expect(component.props().coords).toBeUndefined();
  });
});
