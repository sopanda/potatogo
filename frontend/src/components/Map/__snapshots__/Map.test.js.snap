// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map component tests should render correctly with props 1`] = `
<Styled(Map)
  animate={true}
  attributionControl={true}
  center={
    Array [
      50.041187,
      21.999121,
    ]
  }
  doubleClickZoom={true}
  dragging={true}
  easeLinearity={0.35}
  scrollWheelZoom={true}
  zoom={15}
  zoomControl={true}
>
  <ForwardRef(Leaflet(TileLayer))
    url="https://maps.wikimedia.org/osm-intl/{z}/{x}/{y}.png"
  />
  <ForwardRef(Leaflet(Marker))
    position={
      Array [
        50.041187,
        21.999121,
      ]
    }
  >
    <ForwardRef(Leaflet(Popup))>
      Your current location
    </ForwardRef(Leaflet(Popup))>
  </ForwardRef(Leaflet(Marker))>
</Styled(Map)>
`;
